lib/active_model/serializer/type.rb

```diff
@@ -17,7 +17,7 @@ module ClassMethods
         #   class AdminAuthorSerializer < ActiveModel::Serializer
         #     type 'authors'
         def type(type)
-          self._type = type
+          self._type = type ? type.to_s : nil
```

Comment:
type && type.to_s

B mobile phone

> On Feb 15, 2016, at 1:53 AM, Yohan Robert <notifications@github.com> wrote:
> 
> In lib/active_model/serializer/type.rb:
> 
> > @@ -17,7 +17,7 @@ module ClassMethods
> >          #   class AdminAuthorSerializer < ActiveModel::Serializer
> >          #     type 'authors'
> >          def type(type)
> > -          self._type = type
> > +          self._type = type ? type.to_s : nil
> Because of this, we might call serializer.type(nil) and then nil.to_s results in "". I tried self._type = type if type at first but then two specs were failling out of nowhere. It looked like something related to the cache but I could not really figure it out.
> 
> —
> Reply to this email directly or view it on GitHub.
>